activerecord/lib/active_record/connection_adapters/postgresql/oid/uuid.rb

```diff
@@ -14,7 +14,9 @@ def type
           end
 
           def cast(value)
-            value.to_s[ACCEPTABLE_UUID, 0]
+            return nil unless value.is_a?(String)
```

Comment:
This could break existing apps, it isn't tested, but ` UUIDType.new(guid: :"A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11").guid` is worked for a long time.

Reply:
So there is some reliance on `#to_s` for other types which implement that function, for users who use Object UUIDs instead of serialized UUIDs. Thanks for that feedback I wasn't aware, and it wasn't tested. I'll look into the tests and add a case for a UUID type. I always use `SecureRandom.uuid` myself.

Fortunately, `String#to_s` is just an identity function, so there's only minor performance overhead to returning it, no allocations.

Reply:
I'm sorry I misunderstood your meaning. You are referring to an existing test case in `test_acceptable_uuid_regex` which is a passing test in my change. I thought you were referring to some UUID object library like `uuidtools`, which creates objects to represent UUIDs.

This does not appear to be supported by the existing test cases, though it might work. Because I know I have done that in the past and that others might be expecting it to work, I'll add a test to cover such a case.